flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/TaskExecutorResourceUtils.java

```diff
@@ -692,10 +700,16 @@ public static Configuration adjustConfigurationForLocalExecution(final Configura
 
 		final double shuffleFraction = getShuffleMemoryRangeFraction(configuration).fraction;
 		final double managedFraction = getManagedMemoryRangeFraction(configuration).fraction;
+		final double managedOffHeapFraction;
+		if (isManagedMemoryOffHeapFractionExplicitlyConfigured(configuration)) {
+			managedOffHeapFraction = getManagedMemoryOffHeapFraction(configuration);
+		} else {
```

Comment:
could we move these changes to 87b97fca [hotfix][runtime] If not specified, set Total Flink Memory according to JVM free heap memory for MiniCluster. Also, this will be used in production for local execution, I would suggest to add tests.